app/services/policies/policies_list_service.rb

```diff
@@ -61,4 +65,25 @@ def to_h
       @sets.transform_values(&:to_a).as_json
     end
   end
+
+  HTTP_ERRORS = [HTTPClient::BadResponseError, HTTPClient::TimeoutError, HTTPClient::ConnectTimeoutError, HTTPClient::SendTimeoutError, HTTPClient::ReceiveTimeoutError, SocketError].freeze
+  private_constant :HTTP_ERRORS
+
+  def self.apicast_registry_url
+    ThreeScale.config.sandbox_proxy.apicast_registry_url
+  end
+
+  def self.fetch_policies_from_apicast
+    response = ::JSONClient.get(apicast_registry_url)
+    response.body['policies'] if response.ok? or raise_policies_list_error(response.content)
+  rescue *HTTP_ERRORS => error
+    raise_policies_list_error(error)
+  end
+
+  def self.raise_policies_list_error(error)
+    raise PoliciesListServiceError, I18n.t('errors.messages.apicast_not_found', url: apicast_registry_url, error: error)
+  end
+
+  private_class_method :fetch_policies_from_apicast
+  private_class_method :raise_policies_list_error
```

Comment:
Since we're adding so many methods here, I think it's also a good opportunity to refactor this service. Instead of using class methods, let's make them instance methods instead and stop passing .

Suggestion:

```ruby
def self.call
  new(account).call
end

def initialize(account, builtin: true)
  @account = account
  @builtin = builtin
end

def builtin?
  !!@builtin
end

attr_reader :account

def call
  # More or less the same as the current implementation of self.call
end

def fetch_policies_from_apicast
  return unless builtin?
  # ...
end

def policies_from_account # no longer requires `account` to be passed
  # ...
end
```

Reply:
I would rather keep them class methods for this PR. Once it's working and the JIRA closed, I would refactor it. The reason is simply time.. it would require changes in other files and tests.. it's not so trivial for someone like me and I think it's time to ship this fix.

Reply:
@josemigallas What Gui proposed wouldn't require any change in any other method or class for it to work because from outside it only communicates with `self.call`. But anyway, if you prefer not doing it now and leaving it for one of the following PRs in this topic, it is ok for me 👍

Reply:
https://github.com/3scale/porta/pull/765